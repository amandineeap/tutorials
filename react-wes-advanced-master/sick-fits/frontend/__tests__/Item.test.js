import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
    id: '1212112',
    title: 'my item',
    price: 4000,
    description: 'description',
    image: 'dog.jpg',
    largeImage: 'largedog.jpg'
}

describe('<Item/>', () => {
   it('renders and matches the snapshot', () => {
    //    const price = '$50.35';
    //    expect(price).toMatchSnapshot();
        const wrapper = shallow(<ItemComponent item={fakeItem}></ItemComponent>);
        // expect(wrapper).toMatchSnapshot();
        expect(toJSON(wrapper)).toMatchSnapshot();
   })
    // it('renders the image properly', () => {
    //     const wrapper = shallow(<ItemComponent item={fakeItem}></ItemComponent>);
    //     const img = wrapper.find('img');
    //     // console.log(img.debug());
    //     expect(img.props().src).toBe(fakeItem.image);
    //     expect(img.props().alt).toBe(fakeItem.title);
    // });

    // it('renders the pricetag and title properly', () => {
    //     const wrapper = shallow(<ItemComponent item={fakeItem}></ItemComponent>);
    //     // console.log(wrapper.debug());
    //     const PriceTag = wrapper.find('PriceTag');
    //     // console.log(PriceTag.dive().text());
    //     expect(PriceTag.children().text()).toBe('$50');
    //     // expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
    // });

    // it('renders out the buttons properly',  () => {
    //     const wrapper = shallow(<ItemComponent item={fakeItem}></ItemComponent>);
    //     const buttonList = wrapper.find('.buttonList');
    //     expect(buttonList.children()).toHaveLength(3);
    //     expect(buttonList.find('Link')).toHaveLength(1);
    //     expect(buttonList.find('Link')).toBeTruthy();
    //     expect(buttonList.find('Link').exists()).toBe(true);
    //     // console.log(buttonList.debug());
    //     expect(buttonList.find('DeleteItem').exists()).toBe(true);
    // })
});