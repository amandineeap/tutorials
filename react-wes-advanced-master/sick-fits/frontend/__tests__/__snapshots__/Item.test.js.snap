// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="my item"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "1212112",
          },
        }
      }
    >
      <a>
        my item
      </a>
    </Link>
    <PriceTag>
      $40
    </PriceTag>
    <p>
      description
    </p>
    <div
      className="buttonList"
    >
      <Link
        href={
          Object {
            "pathname": "update",
            "query": Object {
              "id": "1212112",
            },
          }
        }
      >
        <a>
          Edit
        </a>
      </Link>
      <AddToCart
        id="1212112"
      />
      <DeleteItem
        id="1212112"
      >
        Delete this Item
      </DeleteItem>
    </div>
  </Title>
</ItemStyles__Item>
`;
